pkg/config/cmd_terminal.go

```diff
@@ -1,69 +0,0 @@
-//
-// Copyright (c) 2019-2021 Red Hat, Inc.
-// This program and the accompanying materials are made
-// available under the terms of the Eclipse Public License 2.0
-// which is available at https://www.eclipse.org/legal/epl-2.0/
-//
-// SPDX-License-Identifier: EPL-2.0
-//
-// Contributors:
-//   Red Hat, Inc. - initial API and implementation
-//
-
-package config
-
-import (
-	"fmt"
-
-	"github.com/devfile/devworkspace-operator/internal/images"
-
-	dw "github.com/devfile/api/v2/pkg/apis/workspaces/v1alpha2"
-
-	"sigs.k8s.io/yaml"
-)
-
-const (
-	// property name for value with yaml for default dockerimage component
-	// that should be provisioned if devfile DOES have redhat-developer/web-terminal plugin
-	// and DOES NOT have any dockerimage component
-	defaultTerminalDockerimageProperty = "devworkspace.default_dockerimage.redhat-developer.web-terminal"
-)
-
-func (wc *ControllerConfig) GetDefaultTerminalDockerimage() (*dw.Component, error) {
```

Comment:
I think e2e test needs to be adapted after these changes:
```
------------------------------
• Failure [1.527 seconds]
[Create OpenShift Web Terminal Workspace]
/home/sleshche/projects/devworkspace-operator/test/e2e/pkg/tests/devworkspaces_tests.go:25
  Check that pod creator can execute a command in the container [It]
  /home/sleshche/projects/devworkspace-operator/test/e2e/pkg/tests/devworkspaces_tests.go:56

  Cannot execute command in the devworkspace container. Error: `exit status 1`. Exec output: `Error from server (BadRequest): container dev is not valid for pod workspaceda33612faf9843c6-5c775dc868-26vs7         
```

Currently, it references plugin from internal registry
https://github.com/devfile/devworkspace-operator/blob/main/samples/web-terminal.yaml#L16

I think the way we should rework it - make it not terminal specific test but restricted access. Then we can pretty any container component and `controller.devfile.io/restricted-access: "true"` annotation

Reply:
Here's hoping I fixed it -- I don't have OpenShift handy to test changes :crossed_fingers: